### PR TITLE
Added mesos slave collector

### PR DESCRIPTION
**Pull request by drolando**
The mesos master collector doesn't give us any information about the
resources available on each slave. Therefore we need a collector that
queries their snapshot API and emits the metrics returned.

### CODE REVIEW

File: src/fullerite/collector/mesos_slave.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+var (
+	getSlaveExternalIP = util.ExternalIP
+	getSlaveMetrics    = (*MesosSlaveStats).getSlaveMetrics
+
+	getSlaveMetricsURL = func(m *MesosSlaveStats, ip string) string {
+		return fmt.Sprintf("http://%s:%d/metrics/snapshot", ip, m.snapshotPort)
+	}
+
+	// All mesos metrics are gauges except the ones in this list
+	mesosSlaveCountersList = map[string]int{
```
Comment (gnufied) [inline src/fullerite/collector/mesos_slave.go]: This doesn't look like used for DI purposes and looks more like something that doesn't changes.

Comment (drolando) [inline src/fullerite/collector/mesos_slave.go]: The variable is not used as DI at all. Do you want me to move it in another var block?

Comment (gnufied) [inline src/fullerite/collector/mesos_slave.go]: yeah, may be even make it a constant if it nobody modifies it?

Comment (drolando) [inline src/fullerite/collector/mesos_slave.go]: I cannot use a map as constant

Comment (gnufied) [inline src/fullerite/collector/mesos_slave.go]: drat. I am sorry. so yeah, lets move it to another var block. My point is - we shouldn't give wrong impression to the reader.

File: src/fullerite/collector/mesos_slave.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	if decodeErr != nil {
+		m.log.Error("Unable to decode mesos metrics JSON: ", decodeErr.Error())
+		return nil
+	}
+
+	return snapshot
+}
+
+// buildMetric creates the metric and set the correct metricType
+func (m *MesosSlaveStats) buildMetric(name string, value float64) metric.Metric {
```
Comment (gnufied) [inline src/fullerite/collector/mesos_slave.go]: Is this being used from somewhere? If yes, I must have missed something.

Comment (drolando) [inline src/fullerite/collector/mesos_slave.go]: You're right! I forgot to update the sendMetric function :)

File: src/fullerite/collector/mesos_slave.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	s := metric.New(name)
+	s.Value = value
+	if _, exists := mesosSlaveCountersList[name]; exists {
```
Comment (tsheasha) [inline src/fullerite/collector/mesos_slave.go]: maybe mesosSlaveCumulativeCountersList

File: src/fullerite/collector/mesos_slave.go (hunk 1 above)
Comment (tsheasha) [inline src/fullerite/collector/mesos_slave.go]: maybe mesosSlaveCumulativeCountersList?

File: src/fullerite/collector/mesos_slave.go (hunk 3 above)
Comment (drolando) [inline src/fullerite/collector/mesos_slave.go]: done

File: src/fullerite/collector/mesos_slave.go (hunk 2 above)
Comment (drolando) [inline src/fullerite/collector/mesos_slave.go]: fixed

File: src/fullerite/collector/mesos_slave.go (hunk 1 above)
Comment (drolando) [inline src/fullerite/collector/mesos_slave.go]: done

**Comment by tsheasha** [comment]
+1

**Comment by gnufied** [comment]
:shipit:

**Pull request merged**